src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java

```diff
@@ -15,54 +15,56 @@
 package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
 
 public class InputUnnecessaryParenthesesCheckSwitchExpression {
-    MathOperation2 tooManyParens(int k) {
-        return switch (k) {
-            case 1 -> {
-                MathOperation2 case5 = (a, b) -> (a + b); // violation
-                yield case5;
-            }
-            case (2) -> { // violation
-                MathOperation2 case6 = (int a, int b) -> (a + b); // violation
-                yield case6;
-            }
-            case 3 -> {
-                MathOperation2 case7 = (int a, int b) -> {
-                    return (a + b); // violation
-                };
-                yield (case7); // violation
-            }
-            default -> {
-                MathOperation2 case8 = (int x, int y) -> {
-                    return (x + y); // violation
-                };
-                yield case8;
-            }
+  MathOp2 tooManyParens(int k) {
+    return switch (k) {
+      case 1 -> {
+        MathOp2 e = (a, b) -> (a + b); // violation '.* parentheses around assignment right.*side'
+        yield e;
+      }
+      case (2) -> { // violation 'Unnecessary parentheses around literal '2''
+        MathOp2 f = (int a, int b) -> (a + b); // violation '.*paren.*around assignment right.*side'
+        yield f;
+      }
+      case 3 -> {
+        MathOp2 g = (int a, int b) -> {
+          return (a + b); // violation 'Unnecessary parentheses around assignment right-hand side'
         };
-    }
+        yield (g); // violation 'Unnecessary parentheses around identifier 'g''
+      }
+      default -> {
+        MathOp2 h = (int x, int y) -> {
+          return (x + y); // violation 'Unnecessary parentheses around assignment right-hand side'
```

Comment:
>around assignment right-hand side

Assignment ??
Looks like we found a defect in our Check :), it is not assignment.
Please create issue on this.
This problem will not block this is PR, but Issue  need to be created.

Reply:
@rnveach , @pbludov , @strkkk , @nick-mancuso 
Please pay attention to this, good example how good message in input file can help

Reply:
@faraazb , please create issue

Reply:
Please share link to issue here

Reply:
Sorry for taking some time. I tried to understand what may have been causing it. I will create an issue and link it here, very shortly.

Reply:
We can investigate later on, let's not block this PR.

Reply:
Please see #11357

Reply:
Thanks a lot for effort on issue creation